AnnService/inc/Helper/LockFree.h

```diff
@@ -0,0 +1,123 @@
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+
+#ifndef _SPTAG_HELPER_LOCKFREE_H_
+#define _SPTAG_HELPER_LOCKFREE_H_
+
+#include <cstdint>
+#include <vector>
+#include "DiskIO.h"
+
+namespace SPTAG
+{
+    namespace Helper
+    {
+        namespace LockFree
+        {
+            template <typename T>
+            class LockFreeVector 
```

Comment:
the vector improve push_back(emplac_back) latency, but also increase random access(at) latency, have you verify the gain in total?